### ISSUE TITLE
Color profile

### ISSUE THREAD

**Issue by nicolas-p** [factor/factor#1529]
I have never been able to display an image in Factor with the right colors.

In this example, the image was created with OS X's screenshot tool (Cmd+Shift+4). On the left, the image opened in Preview. On the right, the same image in Factor. It's clearly not the same color:

![capture d ecran 2016-02-22 a 20 05 38](https://cloud.githubusercontent.com/assets/11431580/13229312/8da77e9a-d9a0-11e5-8267-609a98949684.png)

I've tried creating images with different color profiles (sRGB, Color LCD, Adobe RBG, Generic RGB), I've tried TIFF and PNG, I've tried converting the image to a different color profile in Preview, nothing worked.

Even the icons in `basis/definitions/icons` (created seven years ago) are not rendered with the same color.

How does Factor render bitmaps? How should I create my images so they end up looking the same in Factor as in Preview?

**Comment by jonenst** [comment]
How about other image viewer programs ? When you screenshot their output, whichcolor does it show ?

**Comment by nicolas-p** [comment]
In the screenshot below, I opened the same image in Autodesk Graphic, Preview, Gimp, Acorn and Factor. I checked with the color picker: only Factor shows a different color.

![capture d ecran 2016-02-22 a 23 45 21](https://cloud.githubusercontent.com/assets/11431580/13235546/c79aaa1e-d9be-11e5-9eeb-543379555dec.png)

It's interesting if you say there's no problem on Ubuntu. There may be something specific to Mac OS X.

**Comment by nicolas-p** [comment]
I did more tests.

With an image created with Mac OS X's screenshot tool:
- on Mac OS X is it displayed **differently** in Factor and Preview
- on Windows it is displayed **differently** in Factor and the image viewer

With an image created with Windows' screenshot tool:
- on Windows it is displayed **identically** in Factor and the image viewer
- on Mac OS X is it displayed **differently** in Factor and Preview

**Comment by bjourne** [comment]
Is there an alpha channel y in the image? If not, then maybe OS X automagically puts Factor in some kind of compatibility mode which messes with the colors?

**Comment by bpollack** [comment]
@nicolas-p Would you mind posting the exact image you've got? I am almost positive that the image has gamma information that Factor's ditching; I'd like to confirm that. If that is the case, the fix may end up being pretty simple (we may just need to load the PNG differently).

**Comment by nicolas-p** [comment]
It was this image:

![v](https://cloud.githubusercontent.com/assets/11431580/13686280/659e38a8-e715-11e5-9f1f-8e3c6babb24b.png)

**Comment by bjourne** [comment]
I'm also seeing it on Linux. Factor, xview and ImageMagick loads the image wrongly, Chromium, Gwenview loads the image right with dampened colors. On Windows, Factor and MsPaint loads it the same but Chrome and Photoviewer loads it dampened. So gamma correction not being loaded is probably a very good guess.

**Comment by erg** [comment]
Im using redshift on linux and f.lux on mac. Relevant?

**Comment by erg** [comment]
http://superuser.com/questions/579216/why-does-this-png-image-display-differently-in-chrome-firefox-than-in-safari-a

**Comment by nicolas-p** [comment]
New test on Mac OS X. Same image created in Graphic with default color parameters (sRGB) and saved in PNG, GIF, JPEG and TIFF. Preview on the left, Factor on the right. The color is different in Factor and Preview every time.

![capture d ecran 2016-03-14 a 18 28 00](https://cloud.githubusercontent.com/assets/11431580/13753415/217be7dc-ea13-11e5-9a21-8f73f1e2916b.png)

**Comment by nicolas-p** [comment]
OK, this is becoming really strange. When I display this page in Safari on a Mac, the three images above have the same color as the Preview side of the screenshot above. But when I display this page in Safari on an iPad, the three images have the same color as the Factor side of the screenshot.

I took a screenshot on the iPad and sent it to the Mac for analysis. The color on the Factor side of the screenshot of the screenshot on the iPad matches the color on the Preview side of the screenshot on the Mac.

**Comment by bjourne** [comment]
It is Factor that is not using the color profile embedded in the image. It's the iCCP chunk. It is not so easy to fix because Factor needs to load the *monitors* color profile, then the one in the image and the image data. Then it needs to combine the two color profiles and the image data to get the right rgb values for the display. 

You can use pngcrush to remove the color profile: `pngcrush -rem iCCP input.png output.png`

**Comment by bjourne** [comment]
Shouldn't be closed. :P

**Comment by mrjbq7** [comment]
We load PNG using the OS image loader, not
Images.png?

**Comment by bjourne** [comment]
True. On Linux we use gdkpixbuf and it doesn't support color management.
